### PR TITLE
Add switchable representation for Packager Status Indicator

### PR DESCRIPTION
**Pull request by JiglioNero**
#1289
Now you can switch representation of Packager Status Indicator in VSCode Status Bar with `react-native.packager.status-indicator` option in `settings.json` to short or full variant. Full variant is default.

|Variant|Indicator representation|
|---|---|
|Full (default)|image|
|Short|image|

### CODE REVIEW

File: src/extension/packagerStatusIndicator.ts [hunk 1]
```diff
@@ -18,48 +19,79 @@ export enum PackagerStatus {
...
+
         this.togglePackagerItem = window.createStatusBarItem(StatusBarAlignment.Left, 10);
-        this.togglePackagerItem.text = `${PackagerStatusIndicator.START_ICON} ${PackagerStatusIndicator.PACKAGER_NAME}`;
-        this.togglePackagerItem.command = PackagerStatusIndicator.START_COMMAND;
-        this.togglePackagerItem.tooltip = localize("StartPackager", "Start Packager");
-        this.togglePackagerItem.show();
+        this.setupPackagerStatusIndicatorItems(PackagerStatusIndicator.START_ICON, PackagerStatusIndicator.START_COMMAND, localize("StartPackager", "Start Packager"));
+    }
+
+    public updateDisplayVersion(){
```
Comment (SounD120) [inline src/extension/packagerStatusIndicator.ts:52]:
```suggestion
    public updateDisplayVersion() {
```

File: src/extension/packagerStatusIndicator.ts [hunk 2]
```diff
@@ -18,48 +19,86 @@ export enum PackagerStatus {
...
     private togglePackagerItem: StatusBarItem;
+    private restartPackagerItem: StatusBarItem;
+    private displayVersion: string = PackagerStatusIndicator.FULL_VERSION;
+    private projectRoot: string|undefined = undefined;
+
     private static PACKAGER_NAME: string = localize("ReactNativePackager", "React Native Packager");
 
     private static START_ICON = "$(play)";
     private static STOP_ICON = "$(primitive-square)";
     private static ACTIVITY_ICON = "$(loading)";
```
Comment (SounD120) [inline src/extension/packagerStatusIndicator.ts:30]: Do we need this as we have Restart icon for now?

File: src/extension/packagerStatusIndicator.ts [hunk 3]
```diff
@@ -18,48 +19,86 @@ export enum PackagerStatus {
...
+    }
+
+    public updateDisplayVersion(): void {
+        let version = PackagerStatusIndicator.FULL_VERSION;
+        try {
+            if (this.projectRoot) {
+                this.displayVersion = SettingsHelper.getPackagerStatusIndicatorPattern(this.projectRoot);
+            }
+        }
+        catch(e) {
```
Comment (SounD120) [inline src/extension/packagerStatusIndicator.ts:59]:
```suggestion
        catch (e) {
```

File: src/extension/packagerStatusIndicator.ts [hunk 4]
```diff
@@ -18,48 +19,86 @@ export enum PackagerStatus {
...
+            this.displayVersion = version;
+        }
     }
 
     public dispose(): void {
         this.togglePackagerItem.dispose();
+        this.restartPackagerItem.dispose();
+    }
+
+    private setupPackagerStatusIndicatorItems(icon: string, command: string|undefined, tooltip: string = ""): void {
```
Comment (SounD120) [inline src/extension/packagerStatusIndicator.ts:71]:
```suggestion
    private setupPackagerStatusIndicatorItems(icon: string, command: string | undefined, tooltip: string = ""): void {
```

File: src/extension/settingsHelper.ts [hunk 5]
```diff
@@ -96,4 +97,20 @@ export class SettingsHelper {
...
+    /**
+     * We get the packager port configured by the user
+     */
+    public static getPackagerStatusIndicatorPattern(fsPath: string): string {
+        const projectRoot = SettingsHelper.getReactNativeProjectRoot(fsPath);
+        let uri = vscode.Uri.file(projectRoot);
+        const workspaceConfiguration = vscode.workspace.getConfiguration("react-native.packager", uri);
+        if (workspaceConfiguration.has("status-indicator")) {
+            const version = ConfigurationReader.readString(workspaceConfiguration.get("status-indicator"));
+            if (version == PackagerStatusIndicator.FULL_VERSION || version == PackagerStatusIndicator.SHORT_VERSION) {
```
Comment (SounD120) [inline src/extension/settingsHelper.ts:110]: Are you sure that `==` compare will work good here? What if we try to pass different types(not string) with the same value?

File: src/extension/packagerStatusIndicator.ts [hunk 6]
```diff
@@ -18,48 +19,86 @@ export enum PackagerStatus {
 
 export class PackagerStatusIndicator implements Disposable {
     private togglePackagerItem: StatusBarItem;
+    private restartPackagerItem: StatusBarItem;
+    private displayVersion: string = PackagerStatusIndicator.FULL_VERSION;
```
Comment (RedMickey) [inline src/extension/packagerStatusIndicator.ts:23]: Do we need initialization here? We can do this in the constructor.

File: src/extension/packagerStatusIndicator.ts [hunk 7]
```diff
@@ -18,48 +19,86 @@ export enum PackagerStatus {
...
 
     private static START_COMMAND = "reactNative.startPackager";
+    private static RESTART_COMMAND = "reactNative.restartPackager";
     private static STOP_COMMAND = "reactNative.stopPackager";
 
-    public constructor() {
+    public static FULL_VERSION = "Full";
+    public static SHORT_VERSION = "Short";
+
+    public constructor(projectRoot: string|undefined = undefined) {
```
Comment (RedMickey) [inline src/extension/packagerStatusIndicator.ts:40]:
```suggestion
    public constructor(projectRoot?: string) {
```

File: src/extension/packagerStatusIndicator.ts [hunk 8]
```diff
@@ -18,48 +19,86 @@ export enum PackagerStatus {
 
 export class PackagerStatusIndicator implements Disposable {
     private togglePackagerItem: StatusBarItem;
+    private restartPackagerItem: StatusBarItem;
+    private displayVersion: string = PackagerStatusIndicator.FULL_VERSION;
+    private projectRoot: string|undefined = undefined;
```
Comment (RedMickey) [inline src/extension/packagerStatusIndicator.ts:24]:
```suggestion
    private projectRoot?: string;
```

File: src/extension/packagerStatusIndicator.ts (hunk 7 above)
Comment (SounD120) [reply src/extension/packagerStatusIndicator.ts:40]: This is not resolved

**Pull request merged**